frontend/src/metabase/admin/performance/utils.ts

```diff
@@ -0,0 +1,129 @@
+import { Cron, weekdays } from "metabase/components/Schedule/constants";
+import type {
+  ScheduleDayType,
+  ScheduleFrameType,
+  ScheduleSettings,
+  ScheduleType,
+} from "metabase-types/api";
+
+const dayToCron = (day: ScheduleSettings["schedule_day"]) => {
+  const index = weekdays.findIndex(o => o.value === day);
+  if (index === -1) {
+    throw new Error(`Invalid day: ${day}`);
+  }
+  return index + 1;
+};
+
+const frameToCron = (frame: ScheduleFrameType) =>
+  ({ first: "1", last: "L", mid: "15" }[frame]);
```

Comment:
define this map (and the ones in the fns below) outside of the function otherwise it gets allocated every call to this function